src/TRestAxionXrayWindow.cxx

```diff
@@ -0,0 +1,318 @@
+}
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestAxionXrayWindow::~TRestAxionXrayWindow() {}
+
+///////////////////////////////////////////////
+/// \brief Initialization of TRestAxionXrayWindow members. It removes all gases.
+///
+void TRestAxionXrayWindow::Initialize() {
+    SetSectionName(this->ClassName());
+    SetLibraryVersion(LIBRARY_VERSION);
+
+    fEnergy.clear();
+    fTransmission.clear();
+}
+
+///////////////////////////////////////////////
+/// \brief It reads the data files from the corresponding material that
+/// needs to be found in the axiolib database with .sol extension.
+/// Usually placed under `data/axion/transmission/`
+///
+void TRestAxionXrayWindow::ReadMaterial() {
+    std::string materialFileName = SearchFile(fMaterial + ".sol");
+
+    debug << "TRestAxionXrayWindow::ReadMaterial. Reading file : " << materialFileName << endl;
+
+    if (!TRestTools::fileExists(materialFileName)) {
+        ferr << "TRestAxionXrayWindow::ReadMaterial( )" << endl;
+        ferr << "Material file not found : " << materialFileName << endl;
+        exit(1);
+    }
+
+    FILE* fin = fopen(materialFileName.c_str(), "rt");
+
+    fEnergy.clear();
+    fTransmission.clear();
+    double en, value;
+    while (fscanf(fin, "%lf\t%lf\n", &en, &value) != EOF) {
+        debug << "Energy : " << en << "eV -- Abs : " << value << endl;
+
+        fEnergy.push_back(en / 1000.);
+        fTransmission.push_back(TMath::Power(value, fThickness * units("um")));
+    }
+
+    debug << "Items read : " << fEnergy.size() << endl;
+
+    fclose(fin);
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the window transmission probability for the given energy (in keV)
+/// and window position, using energy linear interpolation.
+///
+/// For the case of patterned window (stripped or grid), it will return 1 if the strip is
+/// not hitted.
+///
+Double_t TRestAxionXrayWindow::GetTransmission(Double_t energy, Double_t x, Double_t y) {
+    if (fEnergy.size() == 0) ReadMaterial();
+
+    if ((x - fCenter.X()) * (x - fCenter.X()) + (y - fCenter.Y()) * (y - fCenter.Y()) > fRadius * fRadius)
+        return 0;
+
+    if (!HitsPattern(x, y)) return 1.;
+
+    Double_t energyIndex = GetEnergyIndex(energy);
+
+    // Transmission
+    double y2 = fTransmission[energyIndex + 1];
+    double y1 = fTransmission[energyIndex];
+
+    // Normalized field
+    double x2 = fEnergy[energyIndex + 1];
+    double x1 = fEnergy[energyIndex];
+
+    double m = (y2 - y1) / (x2 - x1);
+    double n = y1 - m * x1;
+
+    if (m * energy + n < 0) {
+        ferr << "TRestAxionXrayWindow::GetAbsorptionCoefficient. Negative coefficient!" << endl;
+        cout << "y2 : " << y2 << " y1 : " << y1 << endl;
+        cout << "x2 : " << x2 << " x1 : " << x1 << endl;
+        cout << "m : " << m << " n : " << n << endl;
+        cout << "E : " << energy << " bin : " << energyIndex << endl;
+        GetChar();
+    }
```

Comment:
Is `GetChar` similar to the normal C `getchar`? If so what is this block supposed to be for aside from a debugging help and stopping execution in case this branch happens?